### ISSUE TITLE
Can not include roll formulas

### ISSUE THREAD

**Issue by SabreRunner** [xaukael/inline-chats#1]
Please add a way to include roll formulas inside the tags and not have it break off.

**Comment by xaukael** [comment]
Because I used the same closing tags as an inline roll, it will detect any ]] as the end of the element. Changing the format now would break the module for everyone using it.

**Comment by SabreRunner** [comment]
Can't you detect "[[" and then assume the next "]]" are not closing the inline chat?

**Comment by xaukael** [comment]
not with regex

**Comment by xaukael** [comment]
You are welcome to try of course. Or change the [[! ]] to something else in the code in your instance.

**Comment by Truncated** [comment]
> Can't you detect "[[" and then assume the next "]]" are not closing the inline chat?

can you provide some examples of what you would like to do? Maybe like 3-5 different examples with some reasonable (but actually realistic) complexity? I may be able to work out a regex solution that fits this, but it would have to have some limitations so I need some examples to test it with...

**Comment by xaukael** [comment]
In module.js

Maybe change line 3 to:

`const rgx = /\{\{!(.*?)\}\}/gi;`

and change line 8 to:

`let text = match[0].replace('{{!','').replace('}}','').trim();`

Assuming {{! text }} braces are not used for something else.

You could then do {{! rolled [[1d8]]}}

I have been out of the foundry module dev game for a few months.

**Comment by Truncated** [comment]
one thing I don't know and have to research is what parsing engine is in use here. That could make a big difference - some of the engines out can handle this better (read: at all) than others in the first place.

Additionally, regardless of the first answer, it's extremely likely that only a qualfied option will be workable - i.e. detecting endless number of nested things in all combinations in a single text payload is (almost certainly) not viable, but we may be able to implement a hard limit number of nestings / options for this that covers a lot of expected use.

Still, I'll need some real-life examples of what users want to have work to say for sure - I happen to have done a fair amount of work with regex professionally, but not at all dev for foundry modules nor extensive use of this feature myself.

**Comment by xaukael** [comment]
Okay, I clearly forgot the reasons I knew months ago that would cause this to not work. This method causes an <a> tag (the inline roll) within an <a> tag (the element you click to send it to chat) which does not work.

**Comment by Truncated** [comment]
Welp, if it's not a regex problem, that closes that book for me. Ah well.

**Comment by xaukael** [comment]
ON THE OTHER HAND
if we just use {{ }} on inline rolls rather than [[ ]] we can replace {{ }} with [[ ]] before we process the message.
change line **10**:
```args[0] = args[0].replace(match[0], `<a onclick="ui.chat.processMessage($(this).clone().children().remove().end().text().replace('{{','[[').replace('}}',']]'));" class="inline-chat">${icon}${text}</a>`)```

[[!{{/r 1d8}}]] sends a clickable inline roll to chat, for example

**Comment by Truncated** [comment]
That's a mighty elegant solution you happened upon!
Sometimes just some fresh eyes and a conversation is all that's needed. I'm glad it worked out; I was thinking at the start of ways to parse and detect scenarios with regex, and while that's totally possible to do, I was cringing at the thought of the overhead and potential run-away queries it might lead to...  This is way easier. lol

**Comment by SabreRunner** [comment]
> ON THE OTHER HAND if we just use {{ }} on inline rolls rather than [[ ]] we can replace {{ }} with [[ ]] before we process the message. change line 8: `` args[0] = args[0].replace(match[0], `<a onclick="ui.chat.processMessage($(this).clone().children().remove().end().text().replace('{{','[[').replace('}}',']]'));" class="inline-chat">${icon}${text}</a>`) ``
> 
> [[!{{/r 1d8}}]] sends a clickable inline roll to chat, for example

Does this work? I tried some things on my own with no results.

**Comment by xaukael** [comment]
Yes it works. I did try it before posting it here. What things are you trying with it? It is supposed to be line 10 though. I had the wrong line in the earlier comment.
